dunamai/__init__.py

```diff
@@ -10,9 +10,11 @@
 from enum import Enum
 from functools import total_ordering
 from pathlib import Path
-from typing import Any, Callable, Mapping, Optional, Sequence, Tuple, TypeVar, Union
+from typing import Any, Callable, Mapping, Optional, Sequence, Tuple, TypeVar, Union, NamedTuple
 
-_VERSION_PATTERN = r"^v(?P<base>\d+\.\d+\.\d+)(-?((?P<stage>[a-zA-Z]+)\.?(?P<revision>\d+)?))?$"
+_VERSION_PATTERN = (
+    r"^v(?P<base>\d+\.\d+\.\d+)(-?((?P<stage>[a-zA-Z]+)\.?(?P<revision>\d+)?))?(\+(?P<local>.+))?$"
+)
```

Comment:
Just to make sure I understand the use case here:

* When you tag a repo, you create a tag like `v1.2.3+linux`
* You want the next commit's dynamic version to retain the `+linux` part, like `1.2.3.post1.dev0+linux.g29045e8`

Is that right?

Reply:
yep that is correct